_posts/2018-06-21-vision-for-rust-and-wasm.md

```diff
@@ -0,0 +1,255 @@
+---
+title: Our Vision for Rust and WebAssembly
+---
+
+<meta charset="utf-8" />
+
+**Surgically inserting Rust compiled to WebAssembly should be the best choice
+for speeding up the most performance-sensitive JavaScript code paths.** Do *not*
+throw away your existing code base, because Rust [plays well with
+others][better-for-all]. Regardless if you are a Rust or Web developer, your
+natural workflow shouldn't change because Rust compiled to wasm integrates
+seamlessly into your preferred tools.
+
+This blog post will expand on these aspirations and describe where we stand in
+relation to them right now. In a series of follow up posts, we will talk about
+the next steps for each major component of the Rust and WebAssembly ecosystem.
+
+<style>
+hr {
+    margin-top: 2em;
+    margin-bottom: 2em;
+}
+</style>
+
+--------------------------------------------------------------------------------
+
+> [Are you interested in helping us make these ideals into reality? Join the Rust
+and WebAssembly domain working group!][get-involved]
+
+--------------------------------------------------------------------------------
+
+## Why Focus on Performance-Sensitive Code?
+
+In the most performance-sensitive contexts, JavaScript hinders rather than
+helps. Its dynamic type system and non-deterministic garbage collection pauses
+get in the way. Seemingly small code changes can result in drastic performance
+regressions if you accidentally wander off the JIT's happy path.
+
+On the other hand, Rust gives programmers low-level control and reliable
+performance. It is free from the non-deterministic garbage collection
+pauses. Programmers have control over indirection, monomorphization, and memory
```

Comment:
I don't think a lot of these words mean much as a selling point to a lot of people. I think we can probably simplify this area and the above to something along the lines of "It's hard to write fast JS code without it being unreadable to a lay person. On the other hand it's very easy to write fast Rust code, even if you are new to the language." and then something about how wasm combines this by letting your write fast JS by slipping in wasm under the hood.

Reply:
Not to say that they aren't important, but what sells for you and I might not for another group and so we should tailor it to the right audience. It's kind of like how Rust's tag line isn't a universally selling thing. Who cares about segfaults if all you ever code in is Ruby or JS? You never have that problem there. I think the same goes here.

Reply:
I agree, but also don't want to completely lose this point, because I think it is important to be correct here, and not open ourselves up to attacks based on a misunderstanding. Rust code isn't *faster* per se, and one can write slow code in any language, but it gives you the *control* you need to effectively write fast code, and its idioms push you in that direction.

This is a subtle distinction that doesn't lend itself well to being turned into a sound bite. I'm open to wording suggestions.